tests/unit_tests/c_wrappers/res/config/test_config_content.py

```diff
@@ -74,9 +112,35 @@ def test_config_content_as_dict(tmpdir):
             ConfigKeys.NUM_REALIZATIONS: 42,
             ConfigKeys.QUEUE_OPTION: [["SLURM", "MAX_RUNNING", "50"]],
             "MULTIPLE_KEY_VALUE": [6, 24, 12],
-            ConfigKeys.DATA_FILE: [
-                str(Path.cwd() / existing_file_2),
-                str(Path.cwd() / existing_file_1),
-            ],
+            ConfigKeys.DATA_FILE: str(Path.cwd() / existing_file_2),
             ConfigKeys.REFCASE: str(Path.cwd() / existing_file_1),
         }
+
+
+def test_config_content_as_dict_single_value_keys(tmpdir):
+    with tmpdir.as_cwd():
+        conf = ConfigParser()
+        existing_file_1 = "test_1.t"
+        existing_file_2 = "test_2.t"
+        Path(existing_file_2).write_text("something")
+        Path(existing_file_1).write_text("not important")
+        init_user_config_parser(conf)
+        type_value_map = {
+            "str": "abc",
+            "path": tmpdir,
+            "file_path": existing_file_1,
+            "bool": "TRUE",
+            "float": 4.2,
+            "int": 2,
+        }
+
+        with open("config.file", "w") as fileH:
+            for key in SINGLE_VALUE_KEYS:
+                key_type = SINGLE_VALUE_KEY_TYPES[key]
+                fileH.write(f"{key} {type_value_map.get(key_type, key_type)}\n")
+                fileH.write(f"{key} {type_value_map.get(key_type, key_type)}\n")
```

Comment:
This is probably obvious, but why do you write twice here? Is it to make sure keys are not duplicated at the assert step?

Reply:
Yes, that is precisely why the key value map is written twice in the file.